src/main/webapp/app/shared/service/sort.service.ts

```diff
@@ -95,7 +95,9 @@ export class SortService {
     }
 
     private static customGet(object: any, path: string, defaultValue: any) {
-        const pathArray = path.split('.').filter((key) => key);
+        // Get rid of all optional chainings as they are handled down below. After that split the path into all array and attribute accesses
```

Comment:
```suggestion
        // Get rid of all optional chainings as they are handled down below. After that split the path into all array and attribute accesses
        // Example: path 'some?.path[0].x' will be split into pathArray ['some', 'path', '0', 'x']
```
Small example makes it easy to understand what the overall intention of the replacements and splitting is.